ecosystem/indexer-grpc/indexer-grpc-manager/src/config.rs

```diff
@@ -17,16 +18,32 @@ pub(crate) struct ServiceConfig {
     pub(crate) listen_address: SocketAddr,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct CacheConfig {
+    pub(crate) max_cache_size: usize,
+    pub(crate) target_cache_size: usize,
+}
+
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct IndexerGrpcManagerConfig {
     pub(crate) chain_id: u64,
     pub(crate) service_config: ServiceConfig,
+    #[serde(default = "default_cache_config")]
+    pub(crate) cache_config: CacheConfig,
+    pub(crate) file_store_config: IndexerGrpcFileStoreConfig,
     pub(crate) self_advertised_address: GrpcAddress,
     pub(crate) grpc_manager_addresses: Vec<GrpcAddress>,
     pub(crate) fullnode_addresses: Vec<GrpcAddress>,
 }
 
+const fn default_cache_config() -> CacheConfig {
```

Comment:
why this over impl `Default` trait?

Reply:
Will serde accept that?

Reply:
i thought something like `#[serde(default)]`

Reply:
I am using this on line 32? Isn't this what you mean?